ush/python/pygfs/ufswm/gfs.py

```diff
@@ -1,20 +1,163 @@
 import copy
 import logging
+import os
 
 from pygw.logger import logit
 from pygfs.ufswm.ufs import UFS
 
+from pygw.file_utils import FileHandler
+from pygw.yaml_file import YAMLFile, parse_yamltmpl
+
+
 logger = logging.getLogger(__name__.split('.')[-1])
 
+# ----
+
+# Define the valid GFS configuration attributes.
+VALID_APP_LIST = ["atm"]
+
+# HRW: Only testing C48 for now.
+VALID_ATMRES_LIST = ["c48"]
+VALID_ATMLEVS_LIST = [64, 128]
+
+# ----
+
 
 class GFS(UFS):
+    """
+    Description
+    -----------
+
+    TBD
+
+    Parameters
+    ----------
+
+    config: AttrDict
+
+        A Python dictionary containing the run-time configuration
+        attributes.
+
+    """
 
     @logit(logger, name="GFS")
-    def __init__(self, config):
+    def __init__(self: UFS, config):
+        """
+        Description
+        -----------
+
+        Creates a new GFS object.
 
+        """
+
+        # Define the base-class attributes.
         super().__init__("GFS", config)
 
-        # Start putting fixed properties of the GFS
-        self.ntiles = 6
+        self.ufs_config.ntiles = 6
+        self.config_dirtrees()
+        self.config_fcstapp()
+
+    @logit(logger)
+    def build_dirtree(self: UFS) -> None:
+        """
+        Description
+        -----------
+
+        This method builds the directory tree and links the relevant
+        files.
+
+        """
+
+        fixedfile_list = [item for item in self.ufs_config.fixedfile_list if (
+            "atmos" in item.lower() or "land" in item.lower())]
```

Comment:
This is where a configuration yaml is necessary to elevate this from code to yaml

Reply:
Agreed. I tried to keep this simple for the time-being until we are at that point.